lua/nvim-tree/explorer/init.lua

```diff
@@ -659,6 +659,22 @@ function Explorer:get_nodes_by_line(line_start)
   return nodes_by_line
 end
 
+---@param node Node
+function Explorer:dir_up(node)
+  if not node or node.name == ".." then
+    require("nvim-tree.actions.root.change-dir").fn("..")
+  else
+    local cwd = core.get_cwd()
+    if cwd == nil then
+      return
+    end
+
+    local newdir = vim.fn.fnamemodify(utils.path_remove_trailing(cwd), ":h")
+    require("nvim-tree.actions.root.change-dir").fn(newdir)
+    require("nvim-tree.actions.finders.find-file").fn(node.absolute_path)
```

Comment:
Please move these `require`s up to the top of the file. Here's an example you can follow: https://github.com/nvim-tree/nvim-tree.lua/blob/master/lua/nvim-tree/actions/tree/open.lua#L3

Why do this at the start of the file?
* We only want to do it once, for speed
* Any problems with the require are found immediately at startup, instead of later at runtime